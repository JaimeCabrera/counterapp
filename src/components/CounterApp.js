import React, { useState } from 'react'
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

   const [counter, setCounter] = useState(value);

   const handleAdd = (event) => {
      // console.log(event);
      // tambien se puede retornar una funcion
      // setCounter((counter)=>{counter +1})
      setCounter(counter + 1)
   }

   const handleReset = () => {
      setCounter(value)
   }

   const handleSubstract = () => {
      setCounter(counter -1)
   }

   return (
      <div>
         <h1>Contador</h1>
         <h4>{counter}</h4>
         {/* para pasar argumentos{(argumento)=> funcion(arg)} */}
         <button onClick={handleAdd}>Incrementar +1</button>
         <button onClick={handleReset }>Reset</button>
         <button onClick={handleSubstract}>Reducir -1</button>
      </div>
   )
}

export default CounterApp;

CounterApp.propTypes = {
   value:PropTypes.number.isRequired
}