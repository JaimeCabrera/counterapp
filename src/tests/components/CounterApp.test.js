import '@testing-library/jest-dom';
import React from 'react';
import CounterApp from '../../components/CounterApp';
import { shallow } from 'enzyme';

describe('Pruebas en el componente <CounterApp/>', () => {
// let para tener acceso a la variable en todo el codigo
   // el const es solo una variable de scope {}
   let wrapper = shallow(<CounterApp />);
   // sirve para reinicar las varaiables despues de tal prueba
   beforeEach(() => {
      wrapper = shallow(<CounterApp />);
   })

   test('debe mostrar el valor por defecto el counter', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('debe devolver el valor de 100 del counter app ', () => {
      const valor = 100;
      const wrapper = shallow(<CounterApp value={valor} />)
      const valorComponente = wrapper.find('h4').text().trim();
      expect(valorComponente).toBe('100');
   });

   test('debe incrimentar con el boton de +1', () => {
      wrapper.find('button').at(0).simulate('click', {})
      const valorComponente = wrapper.find('h4').text().trim();
      expect(valorComponente).toBe('11');
   });

   test('debe incrimentar con el boton de -1', () => {
      wrapper.find('button').at(2).simulate('click', {})
      const valorComponente = wrapper.find('h4').text().trim();
      expect(valorComponente).toBe('9');
   });

   test('debe de colocar el valor por defecto al presionar reset', () => {
      const valor = 105;
      const wrapper = shallow(<CounterApp value={valor} />);
      wrapper.find('button').at(0).simulate('click', {});
      wrapper.find('button').at(0).simulate('click', {});
      wrapper.find('button').at(1).simulate('click', {});

      const resetTest = wrapper.find('h4').text().trim();
      expect(resetTest).toBe('105');
      
   })
   
   
});