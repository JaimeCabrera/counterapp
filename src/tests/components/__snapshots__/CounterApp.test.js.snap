// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp/> debe mostrar el valor por defecto el counter 1`] = `
<div>
  <h1>
    Contador
  </h1>
  <h4>
    10
  </h4>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Reducir -1
  </button>
</div>
`;
